packages/datetime/src/dateInput.tsx

```diff
@@ -451,12 +448,6 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
         this.safeInvokeInputProp("onKeyDown", e);
     };
 
-    private setInputRef = (el: HTMLElement) => {
-        this.inputRef = el;
-        const { inputProps = {} } = this.props;
-        Utils.safeInvoke(inputProps.inputRef, el);
```

Comment:
why is this ok to remove? it's a documented feature

Reply:
I removed the unused `inputRef` (due to new compiler error -- it was never read), so `inputProps.inputRef` is applied from the `{...htmlInputProps}` spread